examples/siro_sandbox/controllers/gui_controller.py

```diff
@@ -306,10 +331,88 @@ def _update_grasp(self, grasp_object_id, drop_pos):
 
             self._recorder.record("drop_pos", drop_pos)
 
+        elif speed is not None:
+            grasp_mgr = self._get_grasp_mgr()
+            grasp_object_id = grasp_mgr.snap_idx
+            grasp_mgr.desnap()
+            sim = self._env.task._sim
+            rigid_obj = sim.get_rigid_object_manager().get_object_by_id(
+                grasp_object_id
+            )
+            rigid_obj.motion_type = MotionType.DYNAMIC
+            rigid_obj.collidable = True
+            rigid_obj.override_collision_group(
+                self._thrown_object_collision_group
+            )
+            rigid_obj.linear_velocity = speed
+            obj_bb = get_aabb(grasp_object_id, self._env.task._sim)
+            self._last_object_thrown_info = (
+                rigid_obj,
+                max(obj_bb.size_x(), obj_bb.size_y(), obj_bb.size_z()),
+            )
+
+        if self._last_object_thrown_info is not None:
+            grasp_mgr = self._get_grasp_mgr()
+
+            # when the thrown object leaves the hand, update the collisiongroups
+            rigid_obj = self._last_object_thrown_info[0]
+            ee_pos = (
+                self.get_articulated_agent()
+                .ee_transform(grasp_mgr.ee_index)
+                .translation
+            )
+            dist = np.linalg.norm(ee_pos - rigid_obj.translation)
+            if dist >= self._last_object_thrown_info[1]:
+                rigid_obj.override_collision_group(CollisionGroups.Default)
+                self._last_object_thrown_info = None
+
+    def update_pick_pose(self):
+        num_iters = 0
```

Comment:
Note that this line makes lines 378-398 useless. This is used to create an animation when reaching towards an object, and was turned off so that the hand was responsive when using VR. I am fine with just removing all this.